### ISSUE TITLE
Request mapping template bug of the read operation for @auth directive

### ISSUE THREAD

**Issue by tartachyov** [aws-amplify/amplify-cli#9994]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### Amplify CLI Version

7.6.24

### What operating system are you using?

macOS 

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No, did not apply changes.

### Amplify Categories

auth, api

### Amplify Commands

push

## Request mapping template bug of the read operation for GraphQL API @auth directive with group-based access with groupClaims and dynamic groupField

Here is the example of @auth directive:

> type ChatConversation
    @model
    @auth(
        rules: [
            {
                allow: groups
                provider: userPools
                groupClaim: "custom:GQLUserId2"
                groupsField: "canCR"
                operations: [create, read]
            }
        ]
    ) {
    id: ID! @index(name: "chatConversationId")
    name: String
    canCR: [String]!
}

Create method works fine, but read (get, list) doesn't work and only returns "Unauthorized".

### Expected behavior

I don't see any reason why in this case "create" can work, but "read" doesn't

### Reproduction steps

1. Add GraphQL API to the project
2. Add type in schema.graphql containing @auth directive

```
type ChatConversation
    @model
    @auth(
        rules: [
            {
                allow: groups
                provider: userPools
                groupClaim: "custom:GQLUserId"
                groupsField: "canCR"
                operations: [create, read]
            }
        ]
    ) {
    id: ID! @index(name: "chatConversationId")
    name: String
    canCR: [String]!
}
```

3. Also add custom String "GQLUserId" attribute to the Cognito user pool
4. Set the "custom:GQLUserId" value for your user to be `["someId"]`
5. Using AppSync create new ChatConversation

```
mutation MyMutation {
  createChatConversation(input: {canCR: ["dd108c47-3b36-4cf5-b811-73e20fe58fd7"], name: "New conversation"}) {
    id
    name
  }
}
```

6. Using AppSync list newly created items

```
query MyQuery {
  listChatConversations {
    items {
      name
      id
    }
  }
}
```

7. get result

```
{
  "data": {
    "listChatConversations": null
  },
  "errors": [
    {
      "path": [
        "listChatConversations"
      ],
      "data": null,
      "errorType": "Unauthorized",
      "errorInfo": null,
      "locations": [
        {
          "line": 9,
          "column": 3,
          "sourceName": null
        }
      ],
      "message": "Not Authorized to access listChatConversations on type ModelChatConversationConnection"
    }
  ]
}
```

### GraphQL schema(s)

<details>

```graphql
type ChatConversation
    @model
    @auth(
        rules: [
            {
                allow: groups
                provider: userPools
                groupClaim: "custom:GQLUserId"
                groupsField: "canCR"
                operations: [create, read]
            }
        ]
    ) {
    id: ID! @index(name: "chatConversationId")
    name: String
    canCR: [String]!
}
```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

The issue probably is in the Request mapping template of the `QuerylistChatConversationsauth0Function`.

Line 11 - the issue is that custom claim value is not parsed correctly.
```
## [Start] Authorization Steps. **
$util.qr($ctx.stash.put("hasAuth", true))
#set( $isAuthorized = false )
#set( $primaryFieldMap = {} )
#if( $util.authType() == "API Key Authorization" )

#end
#if( $util.authType() == "User Pool Authorization" )
  #if( !$isAuthorized )
    #set( $authFilter = [] )
    #foreach( $group in $util.defaultIfNull($ctx.identity.claims.get("custom:GQLUserId"), []) )
      #if( !$group.isEmpty() )
        $util.qr($authFilter.add({"canCR": { "contains": $group }}))
      #end
    #end
    #if( !$authFilter.isEmpty() )
      $util.qr($ctx.stash.put("authFilter", { "or": $authFilter }))
    #end
  #end
#end
#if( !$isAuthorized && $util.isNull($ctx.stash.authFilter) )
$util.unauthorized()
#end
$util.toJson({"version":"2018-05-29","payload":{}})
## [End] Authorization Steps. **
```

In my case since I don't need an array I applied dirty hotfix by doing this:
1. Change line 11 to `#foreach( $group in [$ctx.identity.claims.get("custom:GQLUserId")] )`
2. Change user attribute custom:GQLUserId to be just `someId`
3. It works fine now by repeatting the "Reproduction steps" afrom above and I can list items:
```
{
  "data": {
    "listChatConversations": {
      "items": [
        {
          "name": "New conversation",
          "id": "89bbf29e-a809-47ee-bf7f-07b46811a9ae"
        }
      ]
    }
  }
}
```

**Comment by tartachyov** [comment]
@josefaidt yes it looks very similar to #9788 and the origin of it seems to come from [query.ts](https://github.com/aws-amplify/amplify-cli/blob/1496724495010f2daec8b160e35e613ca34eaa5e/packages/amplify-graphql-auth-transformer/src/resolvers/query.ts) file.

Is there any solution better that manually overriding the generated type `...auth` Function?

**Comment by josefaidt** [comment]
Hey @tartachyov thank you for clarifying! And unfortunately I believe that is currently the best workaround. If possible, we can also downgrade the CLI to 7.6.20 per one of the comments from those threads
